xgcm/grid.py

```diff
@@ -1028,41 +1039,66 @@ def __repr__(self):
             summary += axis._coord_desc()
         return "\n".join(summary)
 
-    @docstrings.dedent
-    def interp(self, da, axis, metric_weighted=False, **kwargs):
-        """
-        Interpolate neighboring points to the intermediate grid point along
-        this axis.
+    def _grid_func(self, funcname, da, axis, **kwargs):
+        """this function calls appropriate functions from `Axis` objects.
+        It handles multiple axis input and weighting with metrics
 
         Parameters
         ----------
-        axis : str
-            Name of the axis on which to act
+        axis : {str, list, tuple}
+            Name of the axis on which to act. Multiple axes can be passed as list or
+            tuple (e.g. ['X', 'Y']). Functions will be executed over each axis in the
+            given order.
         metric_weighted : str or tuple of str
             If an axis or list of axes is specified,
             the grid metrics will be used to determined the weight for interpolation.
             If `False` (default), the points will be weighted equally.
+        """
+
+        if (not isinstance(axis, list)) and (not isinstance(axis, tuple)):
+            axis = [axis]
+        # parse multi axis kwargs like e.g. `boundary`
+        multi_kwargs = {k: self._parse_axes_kwargs(v) for k, v in kwargs.items()}
+
+        out = da
+        for axx in axis:
+            kwargs = {k: v[axx] for k, v in multi_kwargs.items()}
+            ax = self.axes[axx]
+            func = getattr(ax, funcname)
+            metric_weighted = kwargs.pop("metric_weighted", False)
+
+            if isinstance(metric_weighted, str):
+                metric_weighted = (metric_weighted,)
+
+            if metric_weighted:
+                metric = self.get_metric(out, metric_weighted)
+                out = out * metric
+
+            out = func(out, **kwargs)
+
+            if metric_weighted:
+                metric_new = self.get_metric(out, metric_weighted)
+                out = out / metric_new
+
+        return out
+
+    @docstrings.dedent
+    def interp(self, da, axis, **kwargs):
+        """
+        Interpolate neighboring points to the intermediate grid point along
+        this axis.
+
+        Parameters
+        ----------
+        %(grid_func.parameters.no_f)s
         %(neighbor_binary_func.parameters.no_f)s
 
         Returns
         -------
         da_i : xarray.DataArray
             The interpolated data
         """
-
-        ax = self.axes[axis]
-
-        if isinstance(metric_weighted, str):
-            metric_weighted = (metric_weighted,)
-
-        if metric_weighted:
-            metric = self.get_metric(da, metric_weighted)
-            da = da * metric
-        out = ax.interp(da, **kwargs)
-        if metric_weighted:
-            metric_new = self.get_metric(out, metric_weighted)
-            out = out / metric_new
-        return out
+        return self._grid_func("interp", da, axis, **kwargs)
```

Comment:
Can we skip writing "interp" and get the function name from context?

I think the answer is no. So it's okay.

Reply:
We probably can. But it doesnt seem that easy. So maybe we stick with a bit of repetition now and check on this later?